libafl/src/inputs/encoded.rs

```diff
@@ -72,14 +92,43 @@ where
 }
 
 impl InputDecoder for TokenInputEncoderDecoder {
-    fn decode(&self, input: &EncodedInput, bytes: &mut Vec<u8>) -> Result<(), Error> {
+    fn decode(&mut self, input: &EncodedInput, bytes: &mut Vec<u8>) -> Result<(), Error> {
+        let mut prev_len = 0;
         for id in input.codes() {
             let tok = self
                 .id_table
                 .get(&(id % self.next_id))
                 .ok_or_else(|| Error::illegal_state(format!("Id {id} not in the decoder table")))?;
+            if self.encoding_type == TokenizationKind::WithWhitespace {
+                let len = tok.len();
+                if prev_len > 1 && len > 1 {
+                    let mut r: u32;
+                    loop {
+                        r = self.rand.below(self.next_id as u64) as u32;
```

Comment:
I don't know if having decoding randomized internay is a good idea, it will be a pain to reproduce. At least store the rand seed in the EncodedJnput struct

Reply:
but that would not be reproducable either? isnt the value of `rand` changed upon calling below()? hence the original value is lost?

Reply:
Yes so the current state should be part of the encoded input, or you want to give a value to `decode`.
Calling `decode` multiple times should never yield different output, else I think a lot of assumptions will break, right? Like, rerunning this input multiple times will yield different results, so the target may behave differently, and you may not be able to reproduce a crash, etc.

Reply:
IMHO decode could result in different decodes. different decodes only happen if there was mutation, never if there was no mutation. so the decode is to be seen as part of the mutation (in afl++ this is in the mutator, in libafl that would be a huge code line change, so putting that in the decoder is much easier and faster.)

Reply:
It should still be reproducible

Reply:
how would it be reproducable if the rand is in the encoded struct?
I could - with a lot of work and time that I do not have now - move it to every single mutation strategy in mutations, slowing things down, and blowing up the code size, but the mutations would also not be reproducable? I do not get the point to be honest.